app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterPresenter.kt

```diff
@@ -28,4 +28,17 @@ class CharacterPresenter(view: CharecterView, private val getChatacterServiceUse
         })
         subscriptions.add(subscription)
     }
+
+    fun requestCharacter(characterId: Int) {
+        val subscription = getChatacterServiceUseCase.invoke(characterId).subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread()).subscribe({ character ->
```

Comment:
use each method invoke in a different line please 
getChatacterServiceUseCase.invoke(characterId)
.subscribeOn(Schedulers.io())
.observeOn(AndroidSchedulers.mainThread())
.subscribe({
......
})

Reply:
in fact, here you have the ID as non-nullable.. so in the use case and service it should be non-nullable too ;)

Reply:
Done